sdk/daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/cctp/MessageSignatureTest.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.data
+package cctp
+
+import java.security.{InvalidKeyException, KeyPairGenerator, SignatureException}
+import java.security.spec.ECGenParameterSpec
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class MessageSignatureTest extends AnyFreeSpec with Matchers {
+
+  "correctly sign and verify secp256k1 signatures" in {
+    val keyPairGen = KeyPairGenerator.getInstance("EC", "BC")
```

Comment:
I'd add a generateKeypair method to MessageSignaturePrototypeUtil so you can avoid the repeated code in this test.

Reply:
From a testing PoV, it works out better to add a keyPairGenerator method